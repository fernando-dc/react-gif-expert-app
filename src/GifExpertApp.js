import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

	// const categories = ['One Punch', 'Samurai X', 'Dragon Ball']
	const [categories, setCategories] = useState(['Dragon Ball']);

	return (
		<>
			<h2  className='animate__animated animate__flipInX animate__slow'>GifExpertApp</h2>
			<AddCategory setCategories= {setCategories}/>
			<hr />

			<ol>
				{
					categories.map((category) => {
						return <GifGrid 
							key={category}
							category={category}
						/>
					})
				}
			</ol>
		</>
	);
}