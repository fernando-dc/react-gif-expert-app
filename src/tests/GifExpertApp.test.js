import React from 'react';

import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Testing for GifExpertApp component', () => {

    test('Must match the snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)
        
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Must show a list of categories', () => {
        const categories = ['Gintama', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    });
    
});
