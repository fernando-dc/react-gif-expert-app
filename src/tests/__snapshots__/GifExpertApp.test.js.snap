// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for GifExpertApp component Must match the snapshot 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__flipInX animate__slow"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Testing for GifExpertApp component Must show a list of categories 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__flipInX animate__slow"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Gintama"
      key="Gintama"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
